library/ssl_tls13_server.c

```diff
@@ -4447,16 +4451,13 @@ int mbedtls_ssl_handshake_server_step( mbedtls_ssl_context *ssl )
 #if defined(MBEDTLS_SSL_TLS13_COMPATIBILITY_MODE)
         case MBEDTLS_SSL_SERVER_CCS_AFTER_SERVER_HELLO:
 
-            /* Only transmit the CCS if we have not done so
-             * earlier already after the HRR.
-             */
-            if( ssl->handshake->hello_retry_requests_sent == 0 )
-                ret = mbedtls_ssl_write_change_cipher_spec( ssl );
+            ret = ssl_write_change_cipher_spec_process(ssl);
```

Comment:
Previously, there was a branch checking that we don't send the CCS multiple times, which is gone now. Why is that? Because any side can send CCS messages as often as they want? Or is there a deeper reason?

Reply:
The check has been moved into ssl_write_change_cipher_spec_coordinate()